ras/perf.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM.
+# Author: Kamalesh Babulal <kamalesh@linux.vnet.ibm.com>
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class perf(Test):
+
+    """
+    Performance analysis tools for Linux
+    :avocado: tags=privileged
+    """
+
+    fail_cmd = list()
+
+    def run_cmd(self, cmd, verbose=True):
+        self.log.info("executing ============== %s =================" % cmd)
+        if process.system(cmd, verbose=verbose, ignore_status=True, sudo=True, shell=True):
+            self.is_fail += 1
+            self.fail_cmd.append(cmd)
+
+    def run_cmd_out(self, cmd):
+        return process.system_output(cmd, shell=True,
+                                     ignore_status=True, sudo=True)
+
+    def setUp(self):
+        smg = SoftwareManager()
+        dist = distro.detect()
+        if 'Ubuntu' in dist.name:
+            linux_tools = "linux-tools-" + os.uname()[2]
+            pkgs = ['linux-tools-common', linux_tools]
+        elif dist.name in ['centos', 'fedora', 'rhel', 'SuSE']:
+            pkgs = ['perf']
+        else:
+            self.cancel("perf is not supported on %s" % dist.name)
+
+        for pkg in pkgs:
+            if not smg.check_installed(pkg) and not smg.install(pkg):
+                self.cancel("Package %s is missing/could not be installed" % pkg)
+
+    def testPerfShort(self):
```

Comment:
minor suggestion: better to have `test_perf_short`. Camel case is usually used for classes. Same for other `test*` methods.